File: C++/KartikeyV/Trees/symmetricTree.cpp
int checkSymmetric(TreeNode *p, TreeNode *q){
    if(p==NULL or q==NULL) return p==q;
    // cout << p->val << ' '<<q->val <<endl;
    return (p->val == q->val and checkSymmetric(p->left,q->right) and checkSymmetric(p->right,q->left));
}

int solve(TreeNode* root){
    return checkSymmetric(root->right, root->left);
}

File: java/ankit_kanyal/devnest_problems/phase_2/trees/SymmetricTree.java
// link- https://devsnest.in/algo-challenges/symmetric-tree?tab=question

static int solve(TreeNode root){
if(Symmetric(root.left,root.right)){
       return 1;
   }
   return 0;
}

public static boolean Symmetric(TreeNode p,TreeNode q){
    if(p==null && q==null) return true;

    if(p!=null && q==null || p==null && q!=null || p.val!=q.val)        return false;

    if(!Symmetric(p.left,q.right)){
        return false;
    }
    if(!Symmetric(p.right,q.left)){
        return false;
    }
    
    return true;
}